overlord/devicestate/devicemgr.go

```diff
@@ -957,11 +958,51 @@ func (m *DeviceManager) Ensure() error {
 	return nil
 }
 
+var errNoSaveSupport = errors.New("no save directory before UC20")
+
+// withSaveDir invokes a function making sure save dir is available.
+// Under UC16/18 it returns errNoSaveSupport
+// For UC20 it also checks that ubuntu-save is available/mounted.
+func (m *DeviceManager) withSaveDir(f func() error) error {
```

Comment:
is it worth exposing this for a unit test for this specific function? we in effect already test it by virtue of using it for the save db and checking the save db, but maybe it would be good to test this directly too? not a blocker if you think it's not worth it

Reply:
maybe in a follow-up. I think the most interesting bit is the errNoSaveSupport return